.github/workflows/basic-ci.yaml

```diff
@@ -23,37 +23,81 @@ jobs:
       matrix:
         containerd_version: ['lts', 'active']
         vmm: ['clh', 'qemu']
-    runs-on: garm-ubuntu-2204-smaller
+        instance: ['garm-ubuntu-2204-smaller', 's390x']
+    runs-on: ${{ matrix.instance }}
     env:
       CONTAINERD_VERSION: ${{ matrix.containerd_version }}
       GOPATH: ${{ github.workspace }}
       KATA_HYPERVISOR: ${{ matrix.vmm }}
     steps:
+      - name: Adjust a permission for repo
+        run: sudo chown -R $USER:$USER $GITHUB_WORKSPACE
+        if: |
+          matrix.instance == 's390x' &&
+          matrix.containerd_version == 'active' && matrix.vmm == 'qemu'
+
+      - name: Take a pre-action for self-hosted runner
+        run: ${HOME}/script/pre_action.sh ubuntu-2204
+        if: |
+          matrix.instance == 's390x' &&
+          matrix.containerd_version == 'active' && matrix.vmm == 'qemu'
+
       - uses: actions/checkout@v4
         with:
           ref: ${{ inputs.commit-hash }}
           fetch-depth: 0
+        if: |
+          matrix.instance == 'garm-ubuntu-2204-smaller' ||
+          ( matrix.containerd_version == 'active' && matrix.vmm == 'qemu' )
 
       - name: Rebase atop of the latest target branch
         run: |
           ./tests/git-helper.sh "rebase-atop-of-the-latest-target-branch"
         env:
           TARGET_BRANCH: ${{ inputs.target-branch }}
+        if: |
+          matrix.instance == 'garm-ubuntu-2204-smaller' ||
+          ( matrix.containerd_version == 'active' && matrix.vmm == 'qemu' )
 
       - name: Install dependencies
         run: bash tests/integration/cri-containerd/gha-run.sh install-dependencies
+        if: |
+          matrix.instance == 'garm-ubuntu-2204-smaller' ||
+          ( matrix.containerd_version == 'active' && matrix.vmm == 'qemu' )
 
-      - name: get-kata-tarball
+      - name: get-kata-tarball-amd64
         uses: actions/download-artifact@v3
         with:
           name: kata-static-tarball-amd64${{ inputs.tarball-suffix }}
           path: kata-artifacts
+        if: matrix.instance == 'garm-ubuntu-2204-smaller'
+
+      - name: get-kata-tarball-s390x
+        uses: actions/download-artifact@v3
+        with:
+          name: kata-static-tarball-s390x${{ inputs.tarball-suffix }}
+          path: kata-artifacts
+        if: |
+          matrix.instance == 's390x' &&
+          matrix.containerd_version == 'active' && matrix.vmm == 'qemu'
 
       - name: Install kata
         run: bash tests/integration/cri-containerd/gha-run.sh install-kata kata-artifacts
+        if: |
+          matrix.instance == 'garm-ubuntu-2204-smaller' ||
+          ( matrix.containerd_version == 'active' && matrix.vmm == 'qemu' )
 
       - name: Run cri-containerd tests
         run: bash tests/integration/cri-containerd/gha-run.sh run
+        if: |
+          matrix.instance == 'garm-ubuntu-2204-smaller' ||
+          ( matrix.containerd_version == 'active' && matrix.vmm == 'qemu' )
+
+      - name: Take a post-action for self-hosted runner
```

Comment:
I think this could be a challenge to achieve the same effect as `always()` under the selective step run. In other words, an unconditional run `always()` conflicts with the conditional below.